src/client/pluginmanager.js

```diff
@@ -128,22 +190,50 @@ define([
          * @param {function} callback
          */
         this.runServerPlugin = function (pluginIdOrClass, context, callback) {
-            var pluginId = typeof pluginIdOrClass === 'string' ? pluginIdOrClass : pluginIdOrClass.metadata.id;
+            var pluginEntry,
+                executionId,
+                pluginId = typeof pluginIdOrClass === 'string' ? pluginIdOrClass : pluginIdOrClass.metadata.id;
             if (context.managerConfig.project instanceof Project) {
                 context.managerConfig.project = context.managerConfig.project.projectId;
             }
 
-            storage.simpleRequest({
-                command: CONSTANTS.SERVER_WORKER_REQUESTS.EXECUTE_PLUGIN,
-                name: pluginId,
-                context: context
-            }, function (err, result) {
-                if (err) {
-                    callback(err, err.result);
-                } else {
-                    callback(null, result);
-                }
-            });
+            getPluginMetadata(pluginId)
+                .then(function (metadata) {
+                    pluginEntry = {
+                        name: metadata.name,
+                        plugin: null,
```

Comment:
Do we need the plugin here?
I think it's better if this data is serializable (but more importantly the same when running inside the browser/server).

Reply:
there is no plugin object here.
in case of server execution we cannot have and we do not need that. we only need the executionId to communicate towards the server...
In case of client execution, it is easier to have the plugin object itself inside this entry, as it is only used in the plugin manager (nobody pass the information around).

Reply:
The plugin-entries are dispatched at `PLUGIN_INITIATED` and `PLUGIN_FINISHED` and also returned at `getRunningPlugins` which makes it a public API. It's fine to store the plugin instances and/or null - but would clean that up before returning them. 
Also please rebase onto master and add the new events to the docs.